backend/src/main/java/com/example/backend/service/LogoutService.java

```diff
@@ -0,0 +1,33 @@
+package com.example.backend.service;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.stereotype.Service;
+
+import com.example.backend.repository.RefreshTokenRepository;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class LogoutService implements LogoutHandler {
+
+    private final RefreshTokenRepository refreshTokenRepository;
+    private final JwtService jwtService;
+
+    @Override
+    public void logout(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            Authentication authentication) {
+
+        String username = jwtService.getUsernameFromRequest(request);
```

Comment:
Dc este nevoie de getUsernameFromRequest?
Nu poate fi folosita functia getCurrentUserUsername?

Reply:
Funcționează fix in acelasi fel, ia token ul de pe request ul care o pornit thread ul